Test 2 add randov products

andmysko opened the pull request.

andmysko committed:
tt

andmysko committed:
test2

andmysko committed:
Merge branch 'master' into Test-2-Add-Randov-Products

andmysko committed:
Test 2

andmysko committed:
test

ovartem commented on src/pages/Inventory.page.js:
```diff
@@ -1,4 +1,5 @@
 /* eslint-disable no-case-declarations */
+import _ from 'lodash';
```
looks like this one was added by mistake in this file.
let's remove it as you don't use it in this particular file

ovartem commented on src/pages/ShoppingCart.page.js:
```diff
@@ -24,4 +30,19 @@ export class ShoppingCartPage extends BaseSwagLabPage {
         await this.cartItems.nth(id).locator(this.removeItemSelector).click();
     }
+
+    async getItemNameById(linkId) {
```
let's rename this method to be more self-explained.
Something like `getItemDataById`

ovartem commented on src/tests/ShoppingCart.spec.js:
```diff
@@ -0,0 +1,40 @@
+        const numbersOfItems = await app.inventory.inventoryItems.count();
+        const linkIds = [];
+        // generate random number
+        for (let i = 0; i < 3; i++) {
```
Let's move this logic to inventory page object

ovartem replied:
Don't hardcode number 3

ovartem commented on src/tests/ShoppingCart.spec.js:
```diff
@@ -0,0 +1,40 @@
+        for (let i = 0; i < 3; i++) {
+            const randomNumber = _.random(0, numbersOfItems - 1);
+            // get title of item
+            const item = app.inventory.itemTitle.nth(randomNumber);
```
Let's simplifyu the logic and iterate over inventory_items (as parent) instead of operating individually item name, desc, price

ovartem commented on src/tests/ShoppingCart.spec.js:
```diff
@@ -0,0 +1,40 @@
+        await app.inventory.shoppingCart.click();
+        // check item title in cart
+        const cartItemDetails = [];
+        for (let i = 0; i < linkIds.length; i++) {
```
It's better to move this code to shoppingCart page object

ovartem reviewed (changes requested):
Really good PR. Even thought it's not working as expected the code looks really solid.

andmysko committed:
fix review
- moved method from tests into page
 - not using the hardcode value
  - avoiding duplicated in random numbers
   - using parent for finding add to cart button

andmysko commented:
@ovartem fixed. pls review

EugenePetrik commented on src/pages/Inventory.page.js:
```diff
@@ -61,4 +65,23 @@ export class InventoryPage extends BaseSwagLabPage {
         }
     }
+
+    async AddRandomItemToTheCart(number, numbersOfItems) {
```
Method name should start from a letter in lower case `addRandomItemToTheCart()`

EugenePetrik commented on src/pages/Inventory.page.js:
```diff
@@ -61,4 +65,23 @@ export class InventoryPage extends BaseSwagLabPage {
+
+    async AddRandomItemToTheCart(number, numbersOfItems) {
+        const randomNumbers = [];
+        const SelectedItems = [];
```
The same for a variable `selectedItems`

EugenePetrik commented on src/pages/ShoppingCart.page.js:
```diff
@@ -7,6 +7,12 @@ export class ShoppingCartPage extends BaseSwagLabPage {
 
     removeItemSelector = '[id^="remove"]';
 
+    cartItemTitle = this.page.locator('[data-test="inventory-item-name"]');
```
We can use `getByTestId` method

```
cartItemTitle = this.page.getByTestId('inventory-item-name');
```

EugenePetrik commented on src/tests/ShoppingCart.spec.js:
```diff
@@ -0,0 +1,30 @@
+        // get numbers of items
+        const numbersOfItems = await app.inventory.inventoryItems.count();
+        // generate how many items to be add to the card
+        const quantityItemsToAdd = _.random(0, numbersOfItems - 1);
```
Based on the task, you should add 2 items to cart, in this case you add all items

EugenePetrik reviewed (commented):
Good job! Left a few comments

andmysko committed:
fix review

andmysko commented:
@EugenePetrik  fixed. pls review

ihorBohdvn commented on src/tests/ShoppingCart.spec.js:
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable playwright/valid-title */
+/* eslint-disable no-plusplus */
+/* eslint-disable no-await-in-loop */
+import { expect } from '@playwright/test';
```
Great job! Few little comments:
Rename the file to a more understandable name, for example, addProductsToCart.spec.js.

ihorBohdvn commented on src/tests/ShoppingCart.spec.js:
```diff
@@ -0,0 +1,19 @@
+import { test } from '../fixtures/base';
+
+test.describe('Verify Shopping Cart', () => {
+    test('Test 2   Add Random Products', async (
```
Change the name of Test 2 Add Random Products to something like Verify adding random products to cart.

andmysko committed:
renamed file and test's name

andmysko commented:
@kirstaVova Thanks. Done. Pls review

EugenePetrik commented on src/pages/Inventory.page.js:
```diff
@@ -9,16 +11,19 @@ export class InventoryPage extends BaseSwagLabPage {
+
     itemPrice = this.page.getByTestId('inventory-item-price');
 
+    parent = this.page.locator('.inventory_item');
```
We can use this

```
this.page.getByTestId('inventory-item');
```

EugenePetrik reviewed (approved):
Great job! Looks good 👍
